cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -196,84 +200,124 @@ func (l *ledgerEntryReadTx) GetLatestLedgerSequence() (uint32, error) {
 	return latestLedgerSeq, err
 }
 
-func (l *ledgerEntryReadTx) getBinaryLedgerEntry(key xdr.LedgerKey) (bool, string, error) {
-	encodedKey, err := encodeLedgerKey(l.buffer, key)
-	if err != nil {
-		return false, "", err
-	}
-
+// From compressed XDR keys to XDR entries (i.e. using the DB's representation)
+func (l *ledgerEntryReadTx) getRawLedgerEntries(keys ...string) (map[string]string, error) {
+	result := make(map[string]string, len(keys))
```

Comment:
should there be a cap here on some sane `len(keys)`?

Reply:
oh I see this is already discussed at length above - feel free to ignore

Reply:
With maps, you only set the cap not the len

Reply:
Ayyy, you mean limit in number l. Yeah, I will do that in a separate PR